spring-kafka/src/main/java/org/springframework/kafka/support/serializer/FailedDeserializationInfo.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.support.serializer;
+
+import java.util.Arrays;
+
+import org.apache.kafka.common.header.Headers;
+
+/**
+ * Class containing all the contextual information around a deserialization error.
+ *
+ * @author Victor Perez Rey
+ *
+ * @since 2.3
+ */
+public class FailedDeserializationInfo {
+
+	private final String topic;
+
+	private final Headers headers;
+
+	private final byte[] data;
+
+	private final boolean isForKey;
+
+	private final Exception exception;
+
+	/**
+	 * Construct an instance with the contextual information.
+	 * @param topic    topic associated with the data.
+	 * @param headers  headers associated with the record; may be empty.
+	 * @param data     serialized bytes; may be null; implementations are recommended to handle null by returning a value or null rather than throwing an exception.
+	 * @param isForKey true for a key deserializer, false otherwise.
+	 * @param exception exception causing the deserialization error.
+	 */
+	public FailedDeserializationInfo(String topic, Headers headers, byte[] data, boolean isForKey, Exception exception) {
+		this.topic = topic;
+		this.headers = headers;
+		this.data = data;
+		this.isForKey = isForKey;
+		this.exception = exception;
+	}
+
+	public String getTopic() {
+		return this.topic;
+	}
+
+	public Headers getHeaders() {
+		return this.headers;
+	}
+
+	public byte[] getData() {
+		return this.data;
+	}
+
+	public boolean isForKey() {
+		return this.isForKey;
+	}
+
+	public Exception getException() {
+		return this.exception;
+	}
+
+	@Override
+	public String toString() {
+		return "FailedDeserializationInfo{" +
+				"topic='" + this.topic + '\'' +
+				", headers=" + headers +
```

Comment:
I think `this.` is missed here.
Please, run `./gradlew clean check --parallel` locally to be sure that you don't violate any Checkstyle rules

Reply:
Funny, locally it passes, also in travis, but that `this` was certainly missing...

Reply:
Hm. I'll check locally.

Well, from big height - LGTM.
I'm pulling your PR locally for final review and some polishing.

Reply:
Well, it looks like Checkstyle has really started to ignore non-overlapping variables where a context is fully clear that we use a property.

Sad, but no choice to live with what we have...